### DISCUSSION TITLE
Make GLOBAL VARIABLES all caps and local variables lower case

### DISCUSSION THREAD

**Discussion by ksdavidc** [ScratchAddons/ScratchAddons#5088] [Ideas]
It is an informal, but useful, tradition? convention? that GLOBAL VARIABLES be all caps and local variables be lower case. I use this convention, BUT for various reasons (alright, I am lazy, that's why) often find I have to convert several at a time that might be of the wrong or mixed type. It seems that it would not be too difficult to implement some aids for this:

Possible implementations:
- simplest, do what it says. It acts on all current variables of both types. (this detects the type)
- separate commands for each type (this detects the type)
- single variable commands that detect the type and CorReCt -> correct it.
- Any or all of the above but without respecting the type (I could make a local all caps, or something like that).

I imagine it as a menu option when right clicking on a variable in the block area (e.g in the swap variables section?), but perhaps it belongs elsewhere (In the variables tab??)
If several options were to be offered, I would suggest having just one menu item, and then a popup or a submenu??? if such a thing is possible.

Admittedly some might argue it only offers occasional or limited improvement the existing situation but I think it could come to be of high value for those who would be apt to use it.

 :)

**Comment by BroJac5246** [comment]
We could also add a little icon, such as the globe emoji next to global variables, for example, sort of like how cloud variables are shown with a cloud. Just an idea :)

**Reply by ksdavidc** [reply]
> We could also add a little icon, such as the globe emoji next to global variables, for example, sort of like how cloud variables are shown with a cloud. Just an idea :)

I like it!

**Reply by matmanna** [reply]
Are you developing this already? If not, I can try to work on it...

**Reply by Norbiros** [reply]
> You can't use emojis in variable names according to **ECMAScript standard, chapter 7.6**  (also it doesn't work in newest firefox)
> https://mothereff.in/js-variables#%F0%9F%8C%8DGLOBAL%5fVARIABLE

NVM, stupid me...

**Reply by mxmou** [reply]
The suggestion is about Scratch variables.

**Reply by Samq64** [reply]
I don't think the Scratch variable names are the ones used when interpreted in JavaScript.

**Reply by Norbiros** [reply]
Um, sorry for that...

**Reply by CST1229** [reply]
> I don't think the Scratch variable names are the ones used when interpreted in JavaScript.

They 100% aren't, the VM uses randomly generated IDs. And also, they are entries in an object so they _might_ be able to contain emojis (because they're strings)?

**Reply by ksdavidc** [reply]
> Are you developing this already? If not, I can try to work on it...

not sure if this was directed at me. not me, go ahead..

**Reply by BroJac5246** [reply]
If you use the backpack to import a cloud variable when you already have 10, it'll keep the cloud icon, but it's actually a Unicode symbol in the name, which can be removed if you rename the variable. Because of that, I'd assume that you can use emojis in variable names (can't verify that right now, though). I'll check soon.

> Are you developing this already? If not, I can try to work on it... 

I'm not, feel free to work on it if you want.

**Reply by BroJac5246** [reply]
> I'd assume that you can use emojis in variable names (can't verify that right now, though). I'll check soon.

Just tested it, and it worked:
![image](https://user-images.githubusercontent.com/104146239/190933296-aef912fe-ba6b-4b30-9b5d-3c9a3f0b1fc0.png)
I saved the project and reloaded it and the variable was still there.
